packages/simcore-sdk/src/simcore_sdk/node_ports_v2/port_validation.py

```diff
@@ -0,0 +1,129 @@
+import logging
+import re
+from typing import Any, Dict, Optional, Tuple
+
+from models_library.projects_nodes import UnitStr
+from models_library.utils.json_schema import (
+    JsonSchemaValidationError,
+    jsonschema_validate_data,
+    jsonschema_validate_schema,
+)
+from pint import PintError, UnitRegistry
+from pydantic.errors import PydanticValueError
+
+JsonSchemaDict = Dict[str, Any]
+
+log = logging.getLogger(__name__)
+
+# ERRORS
+
+#  - Extends PydanticValueError
+#  -> Handled in @validator functions and gathered within ValidationError.errors()
+#  - Use 'code' to discriminate port_validation errors
+
+
+class PortValueError(PydanticValueError):
+    code = "port_validation.schema_error"
+    msg_template = "Invalid value in port {port_key!r}: {schema_error_message}"
+
+    # pylint: disable=useless-super-delegation
+    def __init__(self, *, port_key: str, schema_error: JsonSchemaValidationError):
+        # NOTE: We could extract more info from schema_error (see test_utils_json_schema)
+        # but for the moment we are only interested in the message
+        super().__init__(
+            port_key=port_key,
+            schema_error_path=schema_error.absolute_path,
```

Comment:
Not necessarily. It will be added in the ``ctx`` dict and I use it in https://github.com/ITISFoundation/osparc-simcore/blob/bc6fa1ecae80a94ee3a295c52c6300ed08601a91/services/director-v2/src/simcore_service_director_v2/core/errors.py#L199